File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        for i in range(n):
            l, r = i + 1, n - 1
            temp = target - numbers[i]
            while l <= r:
                mid = (l + r) // 2
                if numbers[mid] > temp:
                    r = mid - 1
                elif numbers[mid] < temp:
                    l = mid + 1
                else:
                    return [i + 1, mid + 1]

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int n = s.size(), m = t.size();
        int i = 0, j = 0;
        while (i < n) {
            if (s[i] == t[j]) {
                i++;
                j++;
            } else {
                i++;
            }
        }
        return m - j;
    }
};